### DISCUSSION TITLE
Streaming the Reply to Save Time

### DISCUSSION THREAD

**Discussion by April-June-August** [chrislemke/ChatFred#93] [Ideas]
**Is your feature request related to a problem? Please describe.**
It can be a bit boring and a time loss waiting for ChatGPT to reply when it got a lot to say. 

**Describe the solution you'd like**
The reply being streamed to Alfred while being generated with `stream` set to `True` for `openai.ChatCompletion.create`. But AFAIK there's no way for an Alfred action like the current Large Type to update while a script being executed by Run Script. So it might have to be implemented using a Python GUI programming tool. 

**Describe alternatives you've considered**
AppleScript dialog, but it can't be updated or closed by itself. Tkinter, but the window won't become the front-most one automatically. 

**Additional context**
I have implemented it using [Fled](https://flet.dev) for myself:

https://github.com/chrislemke/ChatFred/assets/33798984/665c84ae-ae11-4a7a-8398-79c8779f0407

But I think it would be a little invasive for an Alfred Workflow to just launch an app you never installed…

**Comment by April-June-August** [comment]
I have updated the code for this on my fork:

https://github.com/Sponge-bink/ChatFred/blob/12504682df50e6a1b2ff147903ffe76be321fb0e/workflow/src/text_chat_flet.py#L292-L346

**Comment by chrislemke** [comment]
Hey @Sponge-bink cool idea! Please feel free to create a PR!!!

**Comment by April-June-August** [comment]
The PR is live here at #94 !

**Comment by chrislemke** [comment]
I just released a version including this feature:
https://github.com/chrislemke/ChatFred/releases/tag/v1.6.0

Thanks @Sponge-bink!
